lasagne/layers/merge.py

```diff
@@ -275,9 +274,8 @@ def get_output_shape_for(self, input_shapes):
                         for sizes in zip(*input_shapes)]
 
         def match(shape1, shape2):
-            axis = self.axis if self.axis >= 0 else len(shape1) + self.axis
             return (len(shape1) == len(shape2) and
-                    all(i == axis or s1 is None or s2 is None or s1 == s2
+                    all(i == self.axis or s1 is None or s2 is None or s1 == s2
```

Comment:
This will break the tests, you're removing support for a negative axis this way.